engine/src/test/java/org/kigalisim/engine/serializer/EngineResultSerializerTest.java

```diff
@@ -0,0 +1,310 @@
+/**
+ * Unit tests for the EngineResultSerializer class.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.kigalisim.engine.serializer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kigalisim.engine.Engine;
+import org.kigalisim.engine.number.EngineNumber;
+import org.kigalisim.engine.state.ConverterStateGetter;
+
+/**
+ * Tests for the EngineResultSerializer class.
+ */
+public class EngineResultSerializerTest {
```

Comment:
Please validate that no tests / test logic was lost from the JavaScript equivalent found in `test_engine_serializer.js`. If required, have the test create a full EngineResult and have it checked for main body and import supplement separately.